src/collections/componentSet.ts

```diff
@@ -168,23 +171,62 @@ export class ComponentSet extends LazyCollection<MetadataComponent> {
   public static async fromManifest(input: string | FromManifestOptions): Promise<ComponentSet> {
     const manifestPath = typeof input === 'string' ? input : input.manifestPath;
     const options = (typeof input === 'object' ? input : {}) as Partial<FromManifestOptions>;
+    let manifestPre;
+    let manifestPost;
+    const hasManifestPre = !!options.destructivePre;
+    const hasManifestPost = !!options.destructivePost;
 
     const manifestResolver = new ManifestResolver(options.tree, options.registry);
     const manifest = await manifestResolver.resolve(manifestPath);
+    if (hasManifestPre) {
+      manifestPre = await manifestResolver.resolve(options.destructivePre);
```

Comment:
suggest

```ts
const manifestPre = options.destructivePost ? await manifestResolver.resolve(options.destructivePre) : undefined;
```
then you could even refer to the truthiness of `manifestPre` further down

it's cleaner if everything can stay `const`